src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubAppCredentials.java

```diff
@@ -163,18 +196,19 @@ public ListBoxModel doFillApiUriItems() {
         public FormValidation doTestConnection(
                 @QueryParameter("appID") final String appID,
                 @QueryParameter("privateKey") final String privateKey,
-                @QueryParameter("apiUri") final String apiUri
+                @QueryParameter("apiUri") final String apiUri,
+                @QueryParameter("owner") final String owner
 
         ) {
             GitHubAppCredentials gitHubAppCredential = new GitHubAppCredentials(
                     CredentialsScope.GLOBAL, "test-id-not-being-saved", null,
                     appID, Secret.fromString(privateKey)
             );
             gitHubAppCredential.setApiUri(apiUri);
+            gitHubAppCredential.setOwner(owner);
```

Comment:
Commenting here since the `CredentialsScope.GLOBAL` part is just above this:

Do you know if it's possible for this to work with system scope credentials with the branch source plugin? One issue we ran into with our credentials plugin was making the credentials for an org only available to the builds within that org.

Reply:
It is not possible to use system scope credentials, but you can define credentials on an org folder so they are not accessible outside that folder.